### DISCUSSION TITLE
How to set the default name of -Dskywalking.agent.service_name through spring.application.name

### DISCUSSION THREAD

**Discussion by mitht** [apache/skywalking#12428] [Q&A]
How to make the skywalking Java agent automatically retrieve the spring.application.name configuration of the Spring Boot application and set it as the default value for -Dskywalking.agent.service_name?

**Comment by wu-sheng** [comment]
For now, I think there is no mechanism for that. If `spring.application.name` can be set through system environment variable, then you could share that.
